sql/distinct.go

```diff
@@ -37,6 +37,9 @@ func (*planner) distinct(n *parser.Select, p planNode) planNode {
 		d.columnsInOrder = make([]bool, len(p.Columns()))
 	}
 	for _, p := range p.Ordering() {
+		if p == 0 {
```

Comment:
I changed `scanNode.computeOrdering` so that the output is the same length is the input. If ordering is required for a column that is not part of the output, 0 is returned for that index. This change was required so that `analyzeOrdering` could skip over portions of the index for which an exact match was requested. Consider:

```
  SELECT v FROM t WHERE k = 1 ORDER BY v
```

If there is an index on `(k, v)` it will be selected due to the where clause. `k` is neither an explicit or implicit (added by `planner.sort`) output column so the computed scan ordering is `[]int{0, 1}`. `analyzeOrdering` skips over the 0 due to the information returned by `exactPrefix`. 

This stuff is getting complex. I'm feeling there is some missing bit of abstraction that would make it more comprehensible, though I don't know yet what it is.